spec/services/monthly_district_data_service_spec.rb

```diff
@@ -0,0 +1,72 @@
+      hypertensive_patient
+      result = service.report
+      csv = CSV.parse(result)
+      region_row_index = 3
+      expect(find_in_csv(csv, region_row_index, "#")).to eq("All")
+      expect(find_in_csv(csv, region_row_index, "District")).to eq(region.name)
+      expect(csv[region_row_index][2..6].uniq).to eq([nil])
+      expect(find_in_csv(csv, region_row_index, "Total registrations")).to eq("0")
+      expect(find_in_csv(csv, region_row_index, "Total assigned patients")).to eq("1")
+      expect(find_in_csv(csv, region_row_index, "Lost to follow-up patients")).to eq("0")
+      expect(find_in_csv(csv, region_row_index, "Died all-time")).to eq("0")
+      expect(find_in_csv(csv, region_row_index, "Patients under care")).to eq("1")
+      expect(csv[region_row_index][12..24].uniq).to eq(["0"])
+      expect(find_in_csv(csv, region_row_index, "Patients with BP controlled")).to eq("0")
+      expect(find_in_csv(csv, region_row_index, "Patients with BP not controlled")).to eq("0")
+      expect(find_in_csv(csv, region_row_index, "Patients with a missed visits")).to eq("0")
+      expect(find_in_csv(csv, region_row_index, "Patients with a visit but no BP taken")).to eq("0")
+      expect(csv[region_row_index][30..31].uniq).to eq([nil])
+    end
+
+    it "provides accurate numbers for the facilities" do
+      hypertensive_patient
+      result = service.report
+      csv = CSV.parse(result)
+      facility_row_index = 4
+      expect(find_in_csv(csv, facility_row_index, "#")).to eq("1")
+      expect(find_in_csv(csv, facility_row_index, "District")).to eq(region.name)
+      expect(find_in_csv(csv, facility_row_index, "Facility")).to eq(facility.name)
+      expect(find_in_csv(csv, facility_row_index, "Block")).to eq(facility.block)
+      expect(find_in_csv(csv, facility_row_index, "Active/Inactive (Inactive facilities have 0 BP measures taken)"))
+        .to eq("Inactive")
+      expect(find_in_csv(csv, facility_row_index, "Estimated hypertensive population")).to eq(nil)
+      expect(find_in_csv(csv, facility_row_index, "Total registrations")).to eq("0")
+      expect(find_in_csv(csv, facility_row_index, "Total assigned patients")).to eq("1")
+      expect(find_in_csv(csv, facility_row_index, "Lost to follow-up patients")).to eq("0")
+      expect(find_in_csv(csv, facility_row_index, "Died all-time")).to eq("0")
+      expect(find_in_csv(csv, facility_row_index, "Patients under care")).to eq("1")
+      expect(csv[facility_row_index][12..24].uniq).to eq(["0"])
+      expect(find_in_csv(csv, facility_row_index, "Patients with BP controlled")).to eq("0")
+      expect(find_in_csv(csv, facility_row_index, "Patients with BP not controlled")).to eq("0")
+      expect(find_in_csv(csv, facility_row_index, "Patients with a missed visits")).to eq("0")
+      expect(find_in_csv(csv, facility_row_index, "Patients with a visit but no BP taken")).to eq("0")
+      expect(csv[facility_row_index][30..31].uniq).to eq([nil])
```

Comment:
In addition to checking the correctness of individual values, I think it would be valuable to start adding the "property tests" we spoke about recently. It would help ensuring that the data in the reports is integral.

Some example properties for the report here:
- Patients under care = "Patients with BP controlled" + "Patients with BP not controlled" + "Patients with a missed visits" + "Patients with a visit but no BP taken"
- Registrations = Patients under care + ltfu + dead